### PR TITLE
Convert DSE-backed REST API tests SG v1->v2

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Convert existing DSE-specific REST API integration tests from `testing` (SG v1) into new `sgv2-restapi`

**Which issue(s) this PR fixes**:

N/A

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by tatu-at-datastax** [comment]
Finally ready for review; after this will do another PR to remove 2 last remaining tests that run SGv2/REST/DropWizard and we should be able to drop that implementation!

File: apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/persistence/RestApiV2QMaterializedViewIT.java [hunk 1]
```diff
@@ -81,9 +81,9 @@ public void getRowsFromMV() throws Exception {
     // 14-Sep-2022, tatu: Not sure why but there is a transient issue here wrt timing.
     //   Locally test does not appear to ever fail, but in CI it does, with error suggesting
     //   MV has not been created or Schema metadata not (yet) updated. In absence of
-    //   better solution, let's try simple wait to give it time; 5 seconds should do it.
+    //   better solution, let's try simple wait to give it time; 3 seconds should do it.
 
-    Thread.sleep(5000L);
+    Thread.sleep(3000L);
```
Comment (ivansenic) [inline apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/persistence/RestApiV2QMaterializedViewIT.java:86]: better than `Thread.sleep` is to use `Awaitability`, there should be several examples on usage in docs api..

**Review by ivansenic — APPROVED** [review]

File: apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/persistence/RestApiV2QMaterializedViewIT.java (hunk 1 above)
Comment (tatu-at-datastax) [reply apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/persistence/RestApiV2QMaterializedViewIT.java:86]: Yes if there was a way to check for condition. Would love to use it but here all I get is weird failure (index not yet created etc, or whatever preceding query tried to create)

Comment (tatu-at-datastax) [reply apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/persistence/RestApiV2QMaterializedViewIT.java:86]: I guess I can change it for this particular case tho as query tested is not too complex.

Comment (tatu-at-datastax) [reply apis/sgv2-restapi/src/test/java/io/stargate/sgv2/it/persistence/RestApiV2QMaterializedViewIT.java]: Thank you @ivansenic I forgot it's fine to just have the thing fail and not necessarily have condition to wait. Changed as suggested, thanks!

**Pull request merged**